src/settings-manager/metadata-manager.test.ts

```diff
@@ -53,5 +55,69 @@ describe("settings-manager", () => {
       // Set method should populate cache with updated values, so fetch is never called
       expect(fetchMock).toBeCalledTimes(0);
     });
+
+    describe("Delete metadata method", () => {
+      /**
+       * Ensure no breaking changes introduced
+       */
+      it("Constructs if deleteMetadata param is not passed", () => {
+        const manager = new MetadataManager({
+          fetchMetadata: fetchMock,
+          mutateMetadata: mutateMock,
+        });
+
+        expect(manager).toBeDefined();
+      });
+
+      it("Throws if \"delete\" method is called, but deleteMetadata was not passed to constructor", async () => {
+        const manager = new MetadataManager({
+          fetchMetadata: fetchMock,
+          mutateMetadata: mutateMock,
+        });
+
+        await expect(manager.delete("test")).rejects.toThrowErrorMatchingInlineSnapshot(
+          "\"Delete not implemented. Ensure MetadataManager is configured with deleteMetadata param in constructor\""
+        );
+      });
+
+      it("Calls deleteMetadata constructor param when \"delete\" method called", async () => {
+        const manager = new MetadataManager({
+          fetchMetadata: fetchMock,
+          mutateMetadata: mutateMock,
+          deleteMetadata: deleteMetadataMock,
+        });
+
+        await manager.delete("single-key");
+        await manager.delete(["multiple-key-1", "multiple-key-2"]);
+
+        expect(deleteMetadataMock).toBeCalledTimes(2);
+        /**
+         * Ensure callback is called with array type, even if single string was passed
+         */
+        expect(deleteMetadataMock).toHaveBeenNthCalledWith(1, ["single-key"]);
+        expect(deleteMetadataMock).toHaveBeenNthCalledWith(2, ["multiple-key-1", "multiple-key-2"]);
+      });
+
+      it("Accepts records with key and domain and constructs scoped suffixes", async () => {
+        const manager = new MetadataManager({
+          fetchMetadata: fetchMock,
+          mutateMetadata: mutateMock,
+          deleteMetadata: deleteMetadataMock,
+        });
+
+        await manager.delete({ key: "test", domain: "https://test.com/graphql/" });
+        await manager.delete([
+          { key: "test1", domain: "https://test.com/graphql/" },
+          { key: "test2", domain: "https://test.com/graphql/" },
+        ]);
+
+        expect(deleteMetadataMock).toBeCalledTimes(2);
+        expect(deleteMetadataMock).toHaveBeenNthCalledWith(1, ["test__https://test.com/graphql/"]);
+        expect(deleteMetadataMock).toHaveBeenNthCalledWith(2, [
+          "test1__https://test.com/graphql/",
+          "test2__https://test.com/graphql/",
```

Comment:
```suggestion
          { key: "test1", domain: "https://example.com/graphql/" },
          { key: "test2", domain: "https://example.com/graphql/" },
        ]);

        expect(deleteMetadataMock).toBeCalledTimes(2);
        expect(deleteMetadataMock).toHaveBeenNthCalledWith(1, ["test__https://example.com/graphql/"]);
        expect(deleteMetadataMock).toHaveBeenNthCalledWith(2, [
          "test1__https://example.com/graphql/",
          "test2__https://example.com/graphql/",
```
Not a big deal here, but would rather use `example.com` or our own domain for tests in case fetch client is not mocked - we dont want to ping some random domain from CI.